src/main/java/io/javalin/plugin/openapi/OpenApiHandler.kt

```diff
@@ -11,24 +11,37 @@ import io.swagger.v3.oas.models.OpenAPI
 
 class OpenApiHandler(app: Javalin, val options: OpenApiOptions) : Handler {
     private val handlerMetaInfoList = mutableListOf<HandlerMetaInfo>()
+    private var schema: OpenAPI? = null
 
     init {
-        app.events { it.handlerAdded { handlerInfo -> handlerMetaInfoList.add(handlerInfo) } }
+        app.events {
+            it.handlerAdded { handlerInfo ->
+                handlerMetaInfoList.add(handlerInfo)
+                schema = null
+            }
+        }
     }
 
-    fun createOpenAPISchema(): OpenAPI = JavalinOpenApi.createSchema(CreateSchemaOptions(
+    fun createOpenAPISchema(): OpenAPI = JavalinOpenApi.createSchema(
+        CreateSchemaOptions(
             handlerMetaInfoList = handlerMetaInfoList,
             initialConfigurationCreator = options.initialConfigurationCreator,
             default = options.default,
             modelConverterFactory = options.modelConverterFactory,
             packagePrefixesToScan = options.packagePrefixesToScan
-    ))
+        )
+    )
+
+    @Synchronized
```

Comment:
Why is this synchronized?

Reply:
When this is removed, an attacker can send massively (count: n) requests to the OpenApi Handler and if the schema is not generated, it will generate the schema n times. The schema generation proccess is very expensive (1 second on my laptop). With this approach the schema is guarenteed to be generated only once after the handler list changes.

Reply:
Got it! Could you add a comment explaining this?

Reply:
Yes